ssl/src/main/java/io/crate/protocols/ssl/SslContextProviderFactory.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.protocols.ssl;
+
+import io.crate.plugin.EnterpriseLoader;
+import org.elasticsearch.common.Nullable;
+import org.elasticsearch.common.settings.Settings;
+
+/**
+ * Creates an instance of {@link SslContextProvider}. The factory is used in
+ * bundle with a service provider mechanism. See {@link io.crate.plugin.EnterpriseLoader}.
+ *
+ * @see PostgresWireProtocol
+ * @see PipelineRegistry
+ */
+public class SslContextProviderFactory {
+
+    private static SslContextProvider sslContextProvider;
+
+    @Nullable
+    public static SslContextProvider getInstance(Settings settings) {
```

Comment:
I wonder if we should have this as a non-static method and make the class itself a eager singleton created via guice.

Having a "global" singleton like this is a bit dangerous due to our tests (We spawn multiple nodes within the same VM, and usually we want to have components like this to be unique per node).

Reply:
FTR: This doesn't work out because the `PipelineRegistry` (where this is used) is already used int he `Plugin` classes - so guice injection isn't available.